jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -556,7 +586,7 @@ public Content.Source newContentSource(ByteBufferPool.Sized bufferPool, long off
                 ChunksContentSource newContentSource = new ChunksContentSource(chunks);
                 chunks.forEach(Content.Chunk::release);
                 contentSources.add(newContentSource);
-                return newContentSource;
+                return Content.Source.from(newContentSource, offset, length);
```

Comment:
If the length is short enough that it leaves some chunks totally untouched in the wrapped content source, wouldn't that lead to some buffer leak as the `ContentSourceRange` won't read until EOF?

There should be a test explicitly doing this and using the leak tracking pool.

Reply:
Right, I missed that when `ContentSourceRange` does not read to EOF it fails the wrapped source.